### PR TITLE
Add color box to contract assessment dropdown button & dropdown [finishes #89659872]

### PR DESCRIPTION
**Pull request by bcjordan**
contract_match_colors

### CODE REVIEW

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx [hunk 1]
```diff
@@ -1,5 +1,42 @@
 /** TODO(bjordan): Move this all into StudioApps */
```
Comment (Bjvanminnen) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: bug tracking todo? should there be?

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
 $(window).load(function () {
+  $.widget("custom.coloriconselectmenu", $.ui.selectmenu, {
+    /**
+     * Override the jQuery selectmenu to add a color square icon driven by the
+     * data-color attribute on select elements.
+     * @param ul
+     * @param item
+     * @returns {*}
```
Comment (Bjvanminnen) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: what does returns {*} mean?

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx (hunk 1 above)
Comment (bcjordan) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: Removing comment, this seems somewhat out of date. If this happened it'd be part of a broader migration / not practical as a one-off

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx (hunk 2 above)
Comment (bcjordan) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: Ah, that was auto-added when generating the comment, changed to `{jQuery}`

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+    styleCurrentValue: function () {
+      styleButton(this.element);
+    }
+  });
+
+  function bgColorStyle(color) {
+    return "background-color: " + color;
+  }
+
+  function styleButton(selectElement) {
```
Comment (Bjvanminnen) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: Could use a short comment explaining what it's doing.

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+    var selectedColor = $(selectElement).find("option:selected").attr("data-color");
+    var hiddenSelectMenuBtn = "#" + $(selectElement).attr("id") + "-button .ui-selectmenu-text";
```
Comment (Bjvanminnen) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: I might put this string inside the $(). The variable name makes me expect to have a button object, not a string that is a selector to get a button object.

File: dashboard/app/assets/javascripts/levels/contract_match.js.jsx (hunk 3 above)
Comment (bcjordan) [inline dashboard/app/assets/javascripts/levels/contract_match.js.jsx]: Updated name & added method comment

**Comment by Bjvanminnen** [comment]
Couple nits, but lgtm

**Pull request merged**